### DISCUSSION TITLE
Rapid 2.1 Release and What's New

### DISCUSSION THREAD

**Discussion by Bonkles** [facebook/Rapid#993] [General]
### 2.1 is live!
May 4th saw the release of Rapid v2.0, the culmination of our work rewriting the renderer to be as fast as possible. Today, we released v2.1, the first of a new quarterly release program for Rapid. Rapid 2.1 provides new tools for mappers to visualize missing data in OpenStreetMap. 

### Mapscaping podcast appearance!
We recently had the pleasure of appearing on the [Mapscaping podcast](https://mapscaping.com/podcast/the-rapid-editor/)! In it, we discuss the problems Rapid is trying to solve, our collaboration with Mapillary, and our recent v2 rewrite.

<a href="https://mapscaping.com/podcast/the-rapid-editor/"><img src ="https://github.com/facebook/Rapid/assets/1887955/3a292291-952a-4efb-950b-6ea1472308db" alt="mapscaping podcast logo" width="200"></a>

Release Commitment:
The Rapid team is committing to quarterly point releases going forward.  Rapid v2.1 marks the first release under this new commitment. 2.1 includes two features that will bring more attention to map data gaps: 

- A 3D building renderer in an inset map
- The ability to load and visualize more custom data with protomaps tiles support

### Maps in 3D
First, let’s talk about the 3D Map! 
One of the ways that we can encourage new mapping flows is by providing better feedback to editors. Having building heights be displayed in 3D improves the sense of ‘immediate feedback’ and helps add a little joy to the mapping process. 

<img src="https://github.com/facebook/Rapid/assets/1887955/b997e829-6029-4372-9f7f-f1cf1aa6f6e9" alt="3D buildings popping up out of the 2D basemap" height="200">

In Rapid 2.1, you can access the new 3D map viewer via the Background panel:

![image](https://github.com/facebook/Rapid/assets/1887955/f73e1f37-d16c-4e62-bf88-2204767d7cd7)

![3d-buildings-lincoln-memorial](https://github.com/facebook/Rapid/assets/1887955/3602b0df-39f5-4fda-bfe7-878b458aa023)
_above: Rapid's new 3D Building viewer, built with [MapLibre](https://maplibre.org/projects/maplibre-gl-js/)_!

The viewer supports building parts, `min_height` and `max_height` tags, allowing good visualization of multi-part structures like the Lincoln Memorial (above). **Shown here:** selecting and modifying the max_height tag of one building part, then reverting with undo. 

> Note: This 3D Viewer isn’t intended to be a fully-realized 3D scene (though, if that’s of interest, please let us know by filing a [feature request](https://github.com/facebook/Rapid/issues/new/choose)!). The 3D Map viewer is designed only to show enough to highlight missing height data. Among the things it does not support: materials, colors, or roof shapes.  Finally, if you use `building:levels`, the renderer converts the levels count to a height of 3m per level.

### Protomaps PMTiles Support 
This release also includes a refresh of the custom map data feature in Rapid.  You can now add a  [Protomaps PMTiles](https://protomaps.com/docs/pmtiles) URL as a read-only reference data source.  Simply copy any .pmtiles URL into the custom data screen and pan to an area that has data available - you’ll see clickable, inspectable features appear!

One popular source of data in .pmtiles format is [Brandon Liu’s PMTiles demo](https://github.com/bdon/overture-tiles) of the Overture Map Foundation’s [recently released places data](https://overturemaps.org/overture-maps-foundation-releases-first-world-wide-open-map-dataset/).  This layer allows you to quickly identify locations in OpenStreetMap that may be missing important POIs (points of interest).  The points on this layer can serve as a valuable reference layer for improving OpenStreetMap. 

[Click here](https://rapideditor.org/rapid#background=Bing&data=https://r2-public.protomaps.com/protomaps-sample-datasets/overture-pois.pmtiles&datasets=fbRoads,msBuildings&disable_features=boundaries&map=18.25/40.80999/-74.25934) to check out the Overture Places data right from Rapid v2.1, shown below:

![custom-data-overture-places](https://github.com/facebook/Rapid/assets/38784/3dc35516-b656-45f9-8957-0d7af793e77c)

To use your own Custom Data Layer, open the Map Data panel and click the dots button next to the 'Custom Map Data' section (shown above in the red box). Paste your URL into the text box, and hit OK.   For example: ` https://r2-public.protomaps.com/protomaps-sample-datasets/overture-pois.pmtiles`

### Other features
2.1 also includes some behind-the-scenes improvements in how Rapid’s core systems are started, giving better control to apps that want to build Rapid into their core workflows- examples of this are [MapRoulette](https://maproulette.org/) and [HOT’S Tasking Manager](https://tasks.hotosm.org/).

We also fixed a couple dozen bugs- there’s more on that in our latest [changelog](https://github.com/facebook/Rapid/blob/main/CHANGELOG.md).

### Call-to-action
Make rapid your OSM Editor! Bookmark [rapideditor.org/edit](https://rapideditor.org/edit) and make it your daily mapping tool.

Want to help us develop Rapid? Please come find us in the #rapid_feedback channel on the [OSM US slack](http://osmus.slack.com/).

#### Footnote: Retiring the 'Map with AI' name
_When Rapid (Then called RapiD) debuted in July of 2019, its new killer feature was the AI predicted Roads layer created by Meta.  “Map with AI” was meant to conjure the idea of humans working with AI-derived road data to improve the map. Then in 2020, we added the Microsoft AI building layer, further reinforcing the “Map with AI” name.  Rapid’s value-add wasn’t just the data, though, it was the thoughtful human-first editing flow that required someone to approve each and every edit._

_In 2021 & 2022, we completed two big feature sets; First, we Integrated Rapid with ESRI’s [Community Maps Data Sharing](https://www.esri.com/arcgis-blog/products/arcgis-living-atlas/mapping/community-maps-data-sharing/) API, which mostly contains authoritative data not derived from AI. Second, we rewrote the renderer from scratch, bringing Rapid v2 to the world in May 2023._ 

_We feel that the reasons to use Rapid aren’t strictly AI-related, and so, we decided to retire ‘Map with AI’._

**Comment by mxxcon** [comment]
I think 3d building preview has an oversight or a bug. The perimeter `building=` way is supposed to have values of the maximum `height=` and the highest floor(`building:levels=`). 
It looks like the `building=` way overrides any `building:part=yes` ways. But it's supposed to be other way around.
![image](https://github.com/facebook/Rapid/assets/847140/28e074e2-c0cb-49d5-a7ae-d36861b2de0b)

Here you can see that I selected a section of a building that's smaller than its highest point and it's swallowed by the overall building volume.

from https://wiki.openstreetmap.org/wiki/Key:building:part
> Where a building has been split into parts, a building outline has still to be drawn of the whole building way and should have the tags [building:levels](https://wiki.openstreetmap.org/wiki/Key:building:levels)=* and [height](https://wiki.openstreetmap.org/wiki/Key:height)=*. These values should be the maximum of all the building parts. A building that has a three-story part and a six-story part is still a six-story building.

**Reply by Bonkles** [reply]
@mxxcon can you open an Issue on this topic? ⬆️

**Reply by Bonkles** [reply]
Update: #1062 has been fixed and will be available in the 2.1.1 release.
